syntax = "proto3";

package det_yolov3;
import "common.proto";


service DetYOLOv3 {
    rpc DetFrame (DetFrameRequest) returns (Score) {}
    rpc Detect (DetectionRequest) returns (DetectionOutput) {}
}


message DetFrameRequest {
    common.Image image = 1;
    string name = 2;
    string cls = 3;
}

message Score {
    string res = 1;
}

message DetectionRequest {
    common.Image image = 1;
    string name = 2;
    float threshold = 3;
    repeated string targets = 4;
}

message Element {
    string class_name = 1;
    double confidence = 2;
    int32 x1 = 3;
    int32 y1 = 4;
    int32 x2 = 5;
    int32 y2 = 6;
}

message DetectionOutput {
    repeated Element elements = 1;
}
